### DISCUSSION TITLE
Is it possible to save excel while it is opened?

### DISCUSSION THREAD

**Discussion by jason10880** [qax-os/excelize#1883] [General]
I would like to know if it is possible to save excel while it is opened?
User could select the range, then program will paste the picture in the range, then save the file. (picture could show immediately)

**Comment by jason10880** [comment]
add code:
`
func AddPicture(fileName, picFile string) {
	f, err := excelize.OpenFile(fileName)
	if err != nil {
		panic(err)
	}
	defer func() {
		if err := f.Close(); err != nil {
			panic(err)
	}
	}()

	sheetName := f.GetSheetName(f.GetActiveSheetIndex())
	panes, _ := f.GetPanes(sheetName)
	fmt.Printf("%v\n", panes)

	sqRef := panes.Selection[0].SQRef
	activeCell := panes.Selection[0].ActiveCell

	topLeftCell := strings.Split(sqRef, ":")[0]
	bottomRightCell := strings.Split(sqRef, ":")[1]
	if err := f.MergeCell(sheetName, topLeftCell, bottomRightCell); err != nil {
		panic(err)
	}

	opt := &excelize.GraphicOptions{
		AutoFit: true,
	}

	if err := f.AddPicture(sheetName, activeCell, picFile, opt); err != nil {
		panic(err)
	}

	if err := f.UnmergeCell(sheetName, topLeftCell, bottomRightCell); err != nil {
		panic(err)
	}
	
	if err := f.Save(); err != nil {
		panic(err)
	}
}
`

**Comment by xuri** [comment]
The locally installed spreadsheet application doesn't update render views when the content of the workbook file has been changed. You need to reopen the workbook with these apps.

**Reply by jason10880** [reply]
I am using Excel 2016. 
When f.Save() is executed, it will crash with error msg 
"The process cannot access the file because it is being used by another process"

Not sure it is poissible to save the file without reopen and update immediately.
